pages/implementation/reference/typescript.mdx

```diff
@@ -0,0 +1,224 @@
+- React Native
+- Node.js
+
+### Quickstart
+
+Avo functions usage consists of 4 steps.
+
+[//]: # (TODO For future: Here we can add link to the "Type safe code & unit tests" docs section when that's ready, something like:)
+[//]: # 'TODO Learn more about how Avo Functions can help decrease time spent on implementing analytics and increase data reliability in this guide.'
+
+#### Step 1. Include the Avo file
+
+##### Pull the generated code with the Avo CLI
+
+To get the Avo generated Typescript file you must be a member of an Avo workspace with a JavaScript source. Ask for an invite from a colleague or create a <a href="https://avo.app/onboarding" target="_blank">new workspace</a>
+
+```bash
+$ npm install -g avo
+$ avo login
+$ avo pull --branch my-branch-name
+```
+
+Learn more about <Link passHref href="/implementation/cli"><a>the CLI here</a></Link>.
+
+You can also <Link passHref href="/explore-tracking-plan/download-or-copy-avo-file-manually"><a>download the file manually</a></Link> from your Avo workspace.
+
+#### Step 2. Initialize Avo
+
+Import Avo from the generated file and initialize it by calling the `initAvo` method before tracking
+
+```typescript
+import Avo from './Avo';
+
+Avo.initAvo(
+  { env: Avo.AvoEnv.Dev },
+  [systemProperties],
+  [destinationOptions],
+  [customDestDestination],
+);
+```
+
+The actual parameters depend on your tracking plan setup, see the parameters explanation in <Link passHref href="/implementation/reference/typescript#initavo"><a>the reference below</a></Link>.
+
+#### Step 3. Call Avo functions to track your product usage
+
+Every event in your tracking plan, marked with the "Implement with Avo" checkbox, gets a function in the generated code, named according to the event name, in camelCase.
+
+[//]: # 'TODO Future: Add link here to the "Implement with Avo" doc in "Tracking Plan Management"'
+
+For example, if you have a "Signup Start" event defined like this in Avo:
+
+<center style={{ marginBottom: 16 + 'px' }}>
+  <img
+    style={{ height: 600 + 'px' }}
+    src={require('../../../images/signup-start-ui.png')}
+    alt="Event 'Signup Start' defined in Avo with referral string property and implement with Avo check"
+  />
+</center>
+
+You'll be able to call it like this from the generated code
+
+```typescript
+Avo.signupStart({ referral: 'direct' });
+```
+
+> Notice, that you are not passing the System property with the call. System properties are defined on the init step and then automatically included with all events.
+> You can update the system properties with `setSystemProperties` function.
+
+#### Step 4. Verify the implementation
+
+Use the <Link passHref href="/workspace/tracking-plan/implementation-status"><a>Implementation status</a></Link> in your Avo workspace and the <Link passHref href="/explore-tracking-plan/start-using-visual-debuggers"><a>visual debuggers</a></Link> to verify that your implementation is correct.
+
+### Reference
+
+#### initAvo
+
+```typescript
+Avo.initAvo(
```

Comment:
In the recent docs I went with stating actual method signatures here, like
```
export function initAvo(options: {env: AvoEnv; [webDebugger?: boolean];
  [strict?: boolean]; [noop?: boolean];
  [reportFailureAs?: 'error' | 'warn' | 'log']; [inspector?: AvoInspector];
  [avoLogger?: AvoLogger]},
  [systemProperties: {
    systemProperty0: number;
    systemProperty1: boolean;
    ...}],
  [destinationOptions: any], [CustomDestDestination: CustomDestination]) ```
  
  I think it's more clear in the statically typed languages 🤔